### PR TITLE
feat: Add support for Trunk

### PR DESCRIPTION
**Pull request by TylerJang27**
Adds lsp client for [Trunk](https://docs.trunk.io/check), a meta-linter framework. 

I'm pretty new to emacs lisp, so happy to address any feedback!

### CODE REVIEW

**Comment by TylerJang27** [comment]
Are these CI issues unrelated? Looks like other recent jobs on `master` had the same issue.

File: clients/lsp-trunk.el [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+                             :clientType "emacs"
+                             :clientVersion (symbol-value 'emacs-version)))
+  :notification-handlers (ht ("$/progress" #'ignore))
+  :action-handlers (ht ("trunk.checkDisable" #'lsp-trunk-check-disable)
+                       ("trunk.checkEnable" #'lsp-trunk-check-enable)
+                       ("trunk.openConfigFile" #'lsp-trunk-open-config))
+  :priority -2
+  :add-on? t
+  )
+ )
```
Comment (jcs090218) [inline clients/lsp-trunk.el:85]: Don't leave `)` on the newline.

**Comment by jcs090218** [comment]
> Are these CI issues unrelated? Looks like other recent jobs on `master` had the same issue.

Yes, don't worry about this. :)

**Review by jcs090218 — APPROVED** [review]
LGTM. :)

**Pull request merged**